In [3]:
!which python

'which' is not recognized as an internal or external command,
operable program or batch file.


In [6]:
!pip install pandas numpy matplotlib seaborn


[notice] A new release of pip is available: 25.2 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
import numpy as np

In [5]:
!pip install openpyxl


[notice] A new release of pip is available: 25.2 -> 25.3
[notice] To update, run: python.exe -m pip install --upgrade pip


In [31]:
diccionario = pd.read_csv('../references\diccionario_datos_formato_911.csv')

<>:1: SyntaxWarning: invalid escape sequence '\d'
<>:1: SyntaxWarning: invalid escape sequence '\d'
C:\Users\Jesus\AppData\Local\Temp\ipykernel_28320\2006010933.py:1: SyntaxWarning: invalid escape sequence '\d'
  diccionario = pd.read_csv('../references\diccionario_datos_formato_911.csv')


In [32]:
diccionario.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 156 entries, 0 to 155
Data columns (total 2 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   NOMBRE       156 non-null    object
 1   DESCRIPCION  156 non-null    object
dtypes: object(2)
memory usage: 2.6+ KB


In [35]:
datos_2019_2020 = pd.read_csv('../data/raw/formato_911/formato_911_basica_2019-2020.csv')

In [ ]:
import pandas as pd

# Specify the path to your Excel file
excel_file_path = 'your_excel_file.xlsx'

# Create an ExcelFile object
xls = pd.ExcelFile(excel_file_path)

# Access the sheet_names attribute to get a list of all sheet names
sheet_names_list = xls.sheet_names

# Print the list of sheet names
print(sheet_names_list)

FileNotFoundError: [Errno 2] No such file or directory: 'your_excel_file.xlsx'

In [12]:
datos_2019_2020.head()

entidad       n_entidad  municipio                 n_municipi  localidad  \
0        1  AGUASCALIENTES          3                   CALVILLO         33   
1        1  AGUASCALIENTES          1             AGUASCALIENTES       1097   
2        1  AGUASCALIENTES          6        PABELLÓN DE ARTEAGA         10   
3        1  AGUASCALIENTES          1             AGUASCALIENTES       1025   
4        1  AGUASCALIENTES         11  SAN FRANCISCO DE LOS ROMO         84   

                             n_localidad                  domicilio  \
0                     JALTICHE DE ARRIBA      CALLE RINCON DE ROMOS   
1  HACIENDA SAN MARCOS [FRACCIONAMIENTO]      AVENIDA DE LAS PRESAS   
2                            EL GARABATO  CALLE VENUSTIANO CARRANZA   
3                                POCITOS        AVENIDA JESUS MARIA   
4                           EL CHAPARRAL            NINGUNO NINGUNO   

     clavecct                            n_cct  turno  ... doc_1 doc_2 doc_3  \
0  01DPR0270P        SOR JUANA INES DE LA CRUZ      1  ...     0     0     1   
1  01DPR0733G  RAFAEL FRANCISCO AGUILAR LOMELÍ      1  ...     2     1     1   
2  01DPR0175L                 SALVADOR LUEVANO      1  ...     0     0     0   
3  01DTV0067C      ETV NUM. 67 MIGUEL BARRAGAN      1  ...     3     3     2   
4  01KJN0308E           PREESCOLAR COMUNITARIO      1  ...     0     0     0   

  doc_4 doc_5 doc_6 doc_mg doc_tot  tipo_org_docente         tipo_org_alumnos  
0     0     0     1      2       4        MULTIGRADO               INCOMPLETA  
1     1     1     1      0       7    COMPLEMENTARIA                 COMPLETA  
2     0     0     0      2       2        MULTIGRADO               INCOMPLETA  
3     0     0     0      0       8    COMPLEMENTARIA  NO APLICA EN ESTE NIVEL  
4     0     0     0      1       1        MULTIGRADO               INCOMPLETA  

[5 rows x 156 columns]

In [38]:
for i in range(19,25,1):
    print(f'formato_911_basica_20{i}-20{i+1}.csv')
#datos_2019_2020 = pd.read_csv(f'formato_911_basica_20{i}-20{i+1}.csv')

formato_911_basica_2019-2020.csv
formato_911_basica_2020-2021.csv
formato_911_basica_2021-2022.csv
formato_911_basica_2022-2023.csv
formato_911_basica_2023-2024.csv
formato_911_basica_2024-2025.csv


In [39]:
import pandas as pd
import glob
import os

In [41]:
csv_files = sorted(glob.glob("../data/raw/formato_911/formato_911_basica_*.csv"))
csv_files

['../data/raw/formato_911\\formato_911_basica_2019-2020.csv',
 '../data/raw/formato_911\\formato_911_basica_2020-2021.csv',
 '../data/raw/formato_911\\formato_911_basica_2021-2022.csv',
 '../data/raw/formato_911\\formato_911_basica_2022-2023.csv',
 '../data/raw/formato_911\\formato_911_basica_2023-2024.csv']

In [42]:
dataframes = {}

for file in csv_files:
    # Extraer el nombre del archivo sin extensión
    year = os.path.splitext(os.path.basename(file))[0].replace("formato_911_basica_", "")
    
    # Leer el CSV
    df = pd.read_csv(file)
    
    # Guardar en el diccionario
    dataframes[year] = df

# Ejemplo de acceso
dataframes["2022-2023"].head()

entidad       n_entidad  municipio                 n_municipi  localidad                n_localidad               domicilio    clavecct  \
0        1  AGUASCALIENTES          1             AGUASCALIENTES          1             AGUASCALIENTES  CALLE GENERAL BARRAGAN  01PJN0097M   
1        1  AGUASCALIENTES          1             AGUASCALIENTES          1             AGUASCALIENTES       AVENIDA DEL VALLE  01PPR0161N   
2        1  AGUASCALIENTES          2                   ASIENTOS         19               LA ESPERANZA        CALLE CENZONTLES  01KJN0089I   
3        1  AGUASCALIENTES         11  SAN FRANCISCO DE LOS ROMO          1  SAN FRANCISCO DE LOS ROMO   CALLE 10 DE NOVIEMBRE  01DPR0017W   
4        1  AGUASCALIENTES          1             AGUASCALIENTES          1             AGUASCALIENTES         NINGUNO NINGUNO  01EPR0001U   

                                               n_cct  turno   n_turno    tipo       nivel            subnivel c_caracterizan2  control           subcontrol  \
0                             COLEGIO MÉXICO MODERNO      1  MATUTINO  BASICA  PREESCOLAR  PREESCOLAR GENERAL       NO APLICA  PRIVADO           PARTICULAR   
1                                     INSTITUTO CIMA      1  MATUTINO  BASICA    PRIMARIA    PRIMARIA GENERAL       NO APLICA  PRIVADO           PARTICULAR   
2                             PREESCOLAR COMUNITARIO      1  MATUTINO  BASICA  PREESCOLAR   PREESCOLAR CONAFE       NO APLICA  PÚBLICO              FEDERAL   
3                          J. REFUGIO MIRANDA AGUAYO      1  MATUTINO  BASICA    PRIMARIA    PRIMARIA GENERAL       NO APLICA  PÚBLICO  FEDERAL TRANSFERIDO   
4  CENTRO EDUCATIVO INTEGRAL PARA ALTAS CAPACIDAD...      1  MATUTINO  BASICA    PRIMARIA    PRIMARIA GENERAL       NO APLICA  PÚBLICO              ESTATAL   

  periodo  insc_t  hom_t  muj_t  insc_1  hom_1  muj_1  insc_2  hom_2  muj_2  insc_3  hom_3  muj_3  insc_4  hom_4  muj_4  insc_5  hom_5  muj_5  insc_6  hom_6  \
0   I2223      27     16     11       5      3      2       8      5      3      14      8      6       0      0      0       0      0      0       0      0   
1   I2223      83     47     36      19      9     10      18      8     10      20     12      8      14      9      5       7      6      1       5      3   
2   I2223      21     10     11       0      0      0       8      3      5      13      7      6       0      0      0       0      0      0       0      0   
3   I2223     444    192    252      70     34     36      67     35     32      70     27     43      69     32     37      98     35     63      70     29   
4   I2223      58     34     24       0      0      0       0      0      0      12      9      3      14      5      9      17     11      6      15      9   

   muj_6  t_doc_p_h  t_doc_p_m  tot_doc_p  puericul_h  puericul_m  tot_puericul  asi_edu_h  asi_edu_m  tot_asi_edu  dirc_cg_h  dirc_cg_m  tot_dcg  dirc_sg_h  \
0      0          0          3          3           0           0             0          0          0            0          0          0        0          0   
1      2          1          5          6           0           0             0          0          0            0          0          0        0          0   
2      0          0          1          1           0           0             0          0          0            0          0          0        0          0   
3     41          7          6         13           0           0             0          0          0            0          0          0        0          1   
4      6          2          2          4           0           0             0          0          0            0          0          0        0          1   

   dirc_sg_m  tot_dsg  sub_ge_h  sub_ge_m  tot_subge  sub_aca_h  sub_aca_m  tot_subaca  ...  gpos_mg  ins_t  ing_t  rei_t  ins_1  ing_1  rei_1  ins_2  ing_2  \
0          1        1         0         0          0          0          0           0  ...        0     27     2

In [43]:
for year, df in dataframes.items():
    print(year, df.shape)

2019-2020 (230424, 156)
2020-2021 (228852, 156)
2021-2022 (228804, 156)
2022-2023 (229691, 156)
2023-2024 (231534, 156)


Todos los dataframes tienen 156 columnas, aunque difieren en el número de filas.

In [44]:
# Combinar todos los DataFrames y añadir la columna 'periodo_escolar'
combined_df = pd.concat(
    [df.assign(periodo_escolar=year) for year, df in dataframes.items()],
    ignore_index=True
)

In [45]:
combined_df.info()
combined_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1149305 entries, 0 to 1149304
Columns: 157 entries, entidad to periodo_escolar
dtypes: int64(140), object(17)
memory usage: 1.3+ GB


entidad       n_entidad  municipio                 n_municipi  localidad                            n_localidad                  domicilio    clavecct  \
0        1  AGUASCALIENTES          3                   CALVILLO         33                     JALTICHE DE ARRIBA      CALLE RINCON DE ROMOS  01DPR0270P   
1        1  AGUASCALIENTES          1             AGUASCALIENTES       1097  HACIENDA SAN MARCOS [FRACCIONAMIENTO]      AVENIDA DE LAS PRESAS  01DPR0733G   
2        1  AGUASCALIENTES          6        PABELLÓN DE ARTEAGA         10                            EL GARABATO  CALLE VENUSTIANO CARRANZA  01DPR0175L   
3        1  AGUASCALIENTES          1             AGUASCALIENTES       1025                                POCITOS        AVENIDA JESUS MARIA  01DTV0067C   
4        1  AGUASCALIENTES         11  SAN FRANCISCO DE LOS ROMO         84                           EL CHAPARRAL            NINGUNO NINGUNO  01KJN0308E   

                             n_cct  turno   n_turno    tipo       nivel           subnivel c_caracterizan2  control           subcontrol periodo  insc_t  \
0        SOR JUANA INES DE LA CRUZ      1  MATUTINO  BASICA    PRIMARIA   PRIMARIA GENERAL       NO APLICA  PÚBLICO  FEDERAL TRANSFERIDO   I2019     134   
1  RAFAEL FRANCISCO AGUILAR LOMELÍ      1  MATUTINO  BASICA    PRIMARIA   PRIMARIA GENERAL       NO APLICA  PÚBLICO  FEDERAL TRANSFERIDO   I2019     270   
2                 SALVADOR LUEVANO      1  MATUTINO  BASICA    PRIMARIA   PRIMARIA GENERAL       NO APLICA  PÚBLICO  FEDERAL TRANSFERIDO   I2019      59   
3      ETV NUM. 67 MIGUEL BARRAGAN      1  MATUTINO  BASICA  SECUNDARIA     TELESECUNDARIA       NO APLICA  PÚBLICO  FEDERAL TRANSFERIDO   I2019     204   
4           PREESCOLAR COMUNITARIO      1  MATUTINO  BASICA  PREESCOLAR  PREESCOLAR CONAFE       NO APLICA  PÚBLICO              FEDERAL   I2019       1   

   hom_t  muj_t  insc_1  hom_1  muj_1  insc_2  hom_2  muj_2  insc_3  hom_3  muj_3  insc_4  hom_4  muj_4  insc_5  hom_5  muj_5  insc_6  hom_6  muj_6  \
0     71     63      16      7      9      23     14      9      22     10     12      25     17      8      20      7     13      28     16     12   
1    144    126      66     31     35      45     29     16      43     23     20      39     22     17      39     23     16      38     16     22   
2     27     32      11      3      8       9      4      5      10      5      5       8      4      4      14      7      7       7      4      3   
3    112     92      87     50     37      69     41     28      48     21     27       0      0      0       0      0      0       0      0      0   
4      1      0       0      0      0       1      1      0       0      0      0       0      0      0       0      0      0       0      0      0   

   t_doc_p_h  t_doc_p_m  tot_doc_p  puericul_h  puericul_m  tot_puericul  asi_edu_h  asi_edu_m  tot_asi_edu  dirc_cg_h  dirc_cg_m  tot_dcg  dirc_sg_h  \
0          2          2          4           0           0             0          0          0            0          1          0        1          0   
1          1          6          7           0           0             0          0          0            0          0          0        0          0   
2          0          2          2           0           0             0          0          0            0          0          1        1          0   
3          3          5          8           0           0             0          0          0            0          0          0        0          1   
4          0          1          1           0           0             0          0          0            0          0          0        0          0   

   dirc_sg_m  tot_dsg  sub_ge_h  sub_ge_m  tot_subge  sub_aca_h  sub_aca_m  tot_subaca  ...  ins_t  ing_t  rei_t  ins_1  ing_1  rei_1  ins_2  ing_2  rei_2  \
0          0        0         0         0          0          0          0           0  ...    134    134      0     16     16      0     2

In [46]:
#Verificando los valores de la columna "periodo_escolar"
combined_df['periodo_escolar'].unique()

array(['2019-2020', '2020-2021', '2021-2022', '2022-2023', '2023-2024'],
      dtype=object)

Ahora los 6 archivos .csv están cargados todos en un solo DataFrame, combined_df, y existe una nueva columna identificadora llamada "periodo_escolar".

In [50]:
import os

# Subir un nivel (del notebook a la raíz del proyecto)
output_dir = os.path.join("..", "data", "processed")
os.makedirs(output_dir, exist_ok=True)


In [51]:
# Guardar el DataFrame combinado en CSV dentro de ../data/processed
output_path = os.path.join(output_dir, "sep_datos_tidy.csv")
combined_df.to_csv(output_path, index=False, encoding="utf-8-sig")

print(f"✅ Archivo guardado en: {output_path}")
print(f"Total de filas: {len(combined_df)}")


✅ Archivo guardado en: ..\data\processed\sep_datos_tidy.csv
Total de filas: 1149305
